great_expectations/data_context/types/base.py

```diff
@@ -1173,7 +1173,23 @@ class DataContextConfigSchema(Schema):
     )
     config_variables_file_path = fields.Str(allow_none=True)
     anonymous_usage_statistics = fields.Nested(AnonymizedUsageStatisticsConfigSchema)
-    concurrency = fields.Nested(ConcurrencyConfigSchema)
+    concurrency = fields.Nested(
+        ConcurrencyConfigSchema, required=False, allow_none=True
+    )
+
+    # To ensure backwards compatability, we need to ensure that new options are "opt-in"
+    # If a user has not explicitly configured the value, it will be None and will be wiped by the post_dump hook
+    REMOVE_KEYS_IF_NONE = [
+        "concurrency",  # 0.13.33
+    ]
+
+    @post_dump
```

Comment:
Just confirming - do we want `@post_dump` here or `@pre_dump`?

Reply:
hmm so either one works:
* `@pre_dump` takes the `DataContextConfigSchema` before it is serialized into a dictionary
* `@post_dump` is the resulting dictionary

We could iterate through the attributes of the schema but I feel a bit more comfortable working with dictionaries here. Any thoughts here?

Reply:
Seems totally reasonable! `@post_dump` it is.